src/NHibernate/Hql/Ast/ANTLR/QueryTranslatorImpl.cs

```diff
@@ -292,17 +292,14 @@ public bool CanAddFetchedCollectionToCache
 		{
 			get
 			{
-				foreach (var collectionFetch in CollectionFetches)
+				foreach (FromElement fromElement in CollectionFetches)
```

Comment:
If it is guaranteed that they are all always `FromElement` instances like above code implies, it would be better to retype `CollectionFetches` and do the casting in its computing.

After analyzing the code, indeed, it can only be `FromElement` items.